cairo/ethereum/cancun/state.cairo

```diff
@@ -233,33 +233,20 @@ func set_storage{poseidon_ptr: PoseidonBuiltin*, state: State}(
             ),
         );
 
-        let new_storage_trie_ptr = cast(_storage_trie, felt);
+        let storage_trie_pointer = cast(new_storage_trie, felt);
         hashdict_write{poseidon_ptr=poseidon_ptr, dict_ptr=storage_tries_dict_ptr}(
-            1, &address.value, new_storage_trie_ptr
+            1, &address.value, storage_trie_pointer
         );
 
-        let new_storage_tries_dict_ptr = cast(
-            storage_tries_dict_ptr, AddressTrieBytes32U256DictAccess*
-        );
-
-        tempvar new_storage_tries = MappingAddressTrieBytes32U256(
-            new MappingAddressTrieBytes32U256Struct(
-                dict_ptr_start=storage_tries.value.dict_ptr_start,
-                dict_ptr=new_storage_tries_dict_ptr,
-                original_mapping=storage_tries.value.original_mapping,
-            ),
-        );
-
-        tempvar state = State(
-            new StateStruct(
-                _main_trie=state.value._main_trie,
-                _storage_tries=storage_tries,
-                _snapshots=state.value._snapshots,
-                created_accounts=state.value.created_accounts,
-            ),
-        );
-        return ();
+        tempvar storage_trie_pointer = storage_trie_pointer;
+        tempvar storage_tries_dict_ptr = storage_tries_dict_ptr;
+    } else {
+        tempvar storage_trie_pointer = storage_trie_pointer;
+        tempvar storage_tries_dict_ptr = storage_tries_dict_ptr;
     }
+    let storage_trie_pointer = storage_trie_pointer;
+    let storage_tries_dict_ptr = storage_tries_dict_ptr;
```

Comment:
```suggestion
    tempvar storage_trie_pointer = storage_trie_pointer;
    } else {
        tempvar storage_trie_pointer = storage_trie_pointer;
    }
    let storage_tries_dict_ptr = storage_tries_dict_ptr;
```

Even this should work 
The storage_trie_pointer does not need to be rebinded because it is in [ap - 2] in the 2 branches: it is sent back as an implicit argument hashdict_get and hashdict_write

Reply:
true!